plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/HiveMetastoreTableOperations.java

```diff
@@ -90,8 +91,7 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
                 metastore.replaceTable(database, tableName, table, privileges);
             }
             catch (RuntimeException e) {
-                // CommitFailedException is handled as a special case in the Iceberg library. This commit will automatically retry
-                throw new CommitFailedException(e, "Failed to commit to table %s.%s", database, tableName);
+                throw new CommitStateUnknownException(e);
```

Comment:
Just to double-check: we don't need the same logic here as in `FileMetastoreTableOperations`, i.e. to throw a `CommitFailedException` in case the commit swap unambiguously failed?

Reply:
I think https://github.com/trinodb/trino/pull/14118/files/2e46887b6ea7f08a725c2e3169abc553f35ceaa8#r970003583 answers your question.